pymongo/asynchronous/pool.py

```diff
@@ -394,6 +394,26 @@ def _truncate_metadata(metadata: MutableMapping[str, Any]) -> None:
         metadata["platform"] = plat
     else:
         metadata.pop("platform", None)
+    encoded_size = len(bson.encode(metadata))
+    if encoded_size <= _MAX_METADATA_SIZE:
+        return
+    # 5. Truncate driver info.
+    overflow = encoded_size - _MAX_METADATA_SIZE
+    driver = metadata.get("driver", {})
+    if driver:
+        # Truncate driver version.
+        driver_version = driver.get("version")[:-overflow]
```

Comment:
As currently written we could truncate the version from `'4.2.0|1.2.3'` to something like `'4.'` which is a bit odd. Can we at least keep the pymongo version part in full?